src/main/java/com/conveyal/gtfs/GraphQLController.java

```diff
@@ -49,33 +55,28 @@ public static Object post (Request req, Response response) {
             halt(400, "Malformed JSON");
         }
         // FIXME converting String to JSON nodes and back to string, then re-parsing to Map.
-        String vars = node.get("variables").asText();
+        JsonNode vars = node.get("variables");
         String query = node.get("query").asText();
         return doQuery(vars, query, response);
     }
 
-    private static Object doQuery (String varsJson, String queryJson, Response response) {
+
+    private static Object doQuery (JsonNode varsJson, String queryJson, Response response) {
```

Comment:
The type of `result.toSpecification()` is `Map<String, Object>` so the most specific return type of this function is `Map<String, Object>`.